sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -1716,3 +1717,228 @@ test_suite(
+)
+
+_SIGVERIFY_FAIL_MSG = MSG_TEMPLATE_BFV.format(hex(CONST.BFV.SIGVERIFY.BAD_ENCODED_MSG))
+
+device_id_test_cases = [
+    {
+        "name": "device_id_match",
+        "manifest": {"device_id": _test_device_id},
+        "bitstream": ":sigverify_usage_constraint_test_bitstream",
+        "exit_success": DEFAULT_TEST_SUCCESS_MSG,
+        "exit_failure": _SIGVERIFY_FAIL_MSG,
+    },
+    {
+        "name": "device_id_no_match",
+        "manifest": {"device_id": ["0xbad"] * 8},
+        "bitstream": ":sigverify_usage_constraint_test_bitstream",
+        "exit_success": _SIGVERIFY_FAIL_MSG,
+        "exit_failure": DEFAULT_TEST_SUCCESS_MSG,
+    },
+    {
+        "name": "device_id_family_match",
+        "manifest": {"device_id": _test_device_id[0:2]},
+        "bitstream": ":sigverify_usage_constraint_test_bitstream",
+        "exit_success": DEFAULT_TEST_SUCCESS_MSG,
+        "exit_failure": _SIGVERIFY_FAIL_MSG,
+    },
+    {
+        "name": "device_id_family_no_match",
+        "manifest": {"device_id": ["0xbad"] * 3},
+        "bitstream": ":sigverify_usage_constraint_test_bitstream",
+        "exit_success": _SIGVERIFY_FAIL_MSG,
+        "exit_failure": DEFAULT_TEST_SUCCESS_MSG,
+    },
+]
+
+lc_state_test_cases = [
+    {
+        "name": "lc_state_{}_{}".format(
+            lc_state.lower(),
+            "match" if match_case else "no_match",
+        ),
+        "manifest": {"life_cycle_state": "{}".format(hex(lc_state_val)) if match_case else "0"},
+        # Note: the TEST state in software corresponds to all TEST_* states in
+        # hardware. Therefore, the corresponding OTP image used for the TEST
+        # LCV_SW state is //hw/bitstream:rom_otp_test_unlocked0
+        "bitstream": "//hw/bitstream:rom_otp_{}".format(lc_state.lower() if lc_state != "TEST" else "test_unlocked0"),
+        "exit_success": DEFAULT_TEST_SUCCESS_MSG if match_case else _SIGVERIFY_FAIL_MSG,
+        "exit_failure": _SIGVERIFY_FAIL_MSG if match_case else DEFAULT_TEST_SUCCESS_MSG,
+    }
+    for lc_state, lc_state_val in structs.to_dict(CONST.LCV_SW).items()
+    for match_case in (True, False)
+]
+
+manuf_state_test_cases = [
+    {
+        "name": "manuf_state_{}_{}".format(
+            entity,
+            "match" if match_case else "no_match",
+        ),
+        "manifest": {"manuf_state_{}".format(entity): "{}".format(_test_manuf_states[entity]) if match_case else "0"},
+        "bitstream": ":sigverify_usage_constraint_test_bitstream",
+        "exit_success": DEFAULT_TEST_SUCCESS_MSG if match_case else _SIGVERIFY_FAIL_MSG,
+        "exit_failure": _SIGVERIFY_FAIL_MSG if match_case else DEFAULT_TEST_SUCCESS_MSG,
+    }
+    for entity in ("creator", "owner")
+    for match_case in (True, False)
+]
+
+all_constr_test_cases = [{
+    "name": "all_constraints",
+    "manifest": {
+        "device_id": _test_device_id,
+        "life_cycle_state": "{}".format(hex(CONST.LCV_SW.RMA)),  # TODO: which LC state would be best to test here?
```

Comment:
I'm currently testing against the RMA LC state in the "all constraints" test. Is there another one I should prefer?